Looks fine, but you need to remove the null check in the `GlowPlayerProfile` constructor to make it work properly:

```
10:41:59 [SEVERE] Could not pass event StatusEvent to ServerListPlus v3.4.9-SNAPSHOT
org.bukkit.event.EventException
        ...
Caused by: java.lang.NullPointerException: uuid must not be null
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:787)
        at net.glowstone.entity.meta.profile.GlowPlayerProfile.<init>(GlowPlayerProfile.java:93)
        at net.glowstone.entity.meta.profile.GlowPlayerProfile.<init>(GlowPlayerProfile.java:60)
        at net.glowstone.GlowServer.createProfile(GlowServer.java:1760)
        at net.glowstone.GlowServer.createProfile(GlowServer.java:1748)
        at net.glowstone.GlowServer.createProfile(GlowServer.java:1743)
        at net.glowstone.GlowServer.createProfile(GlowServer.java:215)
        at net.minecrell.serverlistplus.bukkit.handlers.PaperEventHandler.handlePaperServerListPing(PaperEventHandler.java:101)
        at net.minecrell.serverlistplus.bukkit.handlers.PaperEventHandler.onServerListPing(PaperEventHandler.java:28)
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor3.execute(Unknown Source)
        at org.bukkit.plugin.EventExecutor$2.execute(EventExecutor.java:70)
        ... 45 more
```

--- a/src/main/java/net/glowstone/entity/meta/profile/GlowPlayerProfile.java
+++ b/src/main/java/net/glowstone/entity/meta/profile/GlowPlayerProfile.java
@@ -90,7 +90,6 @@ public class GlowPlayerProfile implements PlayerProfile {
     private GlowPlayerProfile(String name, CompletableFuture<UUID> uuid,
             Collection<ProfileProperty> properties) {
         checkNotNull(properties, "properties must not be null");
-        checkNotNull(uuid, "uuid must not be null");
         this.name = name;
         this.uniqueId = uuid;
         this.properties = Maps.newHashMap();

